airbyte-integrations/connectors/source-amazon-seller-partner/source_amazon_seller_partner/streams.py

```diff
@@ -103,13 +103,16 @@ def request_params(
         if next_page_token:
             return dict(next_page_token)
 
-        params = {self.replication_start_date_field: self._replication_start_date, self.page_size_field: self.page_size}
+        params = {}
 
-        if self._replication_start_date and self.cursor_field:
-            start_date = max(stream_state.get(self.cursor_field, self._replication_start_date), self._replication_start_date)
-            params.update({self.replication_start_date_field: start_date})
+        if self.replication_start_date_field is not None:
+            params.update({self.replication_start_date_field: self._replication_start_date, self.page_size_field: self.page_size})
 
-        if self._replication_end_date:
+            if self._replication_start_date and self.cursor_field:
+                start_date = max(stream_state.get(self.cursor_field, self._replication_start_date), self._replication_start_date)
+                params.update({self.replication_start_date_field: start_date})
+
+        if self.replication_end_date_field is not None and self._replication_end_date:
```

Comment:
Any reason why you're doing this change? Can you add a unit test to validate the new logic?

Reply:
@efimmatytsin I'd prefer to keep the original function unchanged and you can override it for the OrderItems case.

Reply:
done.